dom/src/main/scala-3/fs2/dom/Dom.scala

```diff
@@ -48,6 +52,153 @@ object Node {
       node.replaceChild(newChild, oldChild)
       ()
     }
+
+    def addEventListener[T <: Event](`type`: String, options: EventListenerOptions)(using
+        F: Dom[F]
+    ): Resource[F, Stream[F, T]] =
+      EventTargetHelpers.listen(node, `type`, options)
```

Comment:
I know it's making more work, but I was imagining we can do something like this:
```scala
def addEventListener(`type`: "click", ...): Stream[F, MouseEvent[F]]
```

Reply:
Ok, I'm just not a big fan of method overloading as it can result in hard to understand error messages (at least in Scala 2). What about a sealed trait of dom events holding a Event type member plus a corresponding path dependent function?

Reply:
Ok, I'm just not a big fan of method overloading as it can result in hard to understand error messages (at least in Scala 2). What about a sealed trait of dom events holding a Event type member plus a corresponding path dependent function?

Reply:
> Ok, I'm just not a big fan of method overloading as it can result in hard to understand error messages (at least in Scala 2)

That's fair. I'm not sure if it's improved in Scala 3.

> What about a sealed trait of dom events holding a Event type member plus a corresponding path dependent function?

Hmm, I'd have to see this. In my experience path-dependent types are also confusing 😅

Reply:
> Hmm, I'd have to see this. In my experience path-dependent types are also confusing sweat_smile
I can probably PR this tomorrow. I agree, in Scala 2 it has been completely unsound to a bunch of other type features, but my hopes are DOT calculus is helping us out now :smile:

Reply:
@armanbilge see https://github.com/armanbilge/fs2-dom/pull/27/commits/2287b2ed9957d9b47dbec36400f6231114dcb419